src/hkdf.rs

```diff
@@ -0,0 +1,261 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+/*!
+ * This module implements the HMAC-based Extract-and-Expand Key
+ * Derivation Function as specified by  https://tools.ietf.org/html/rfc5869.
+ */
+
+use std::iter::repeat;
+use std::num::Int;
+use std::slice::bytes::copy_memory;
+
+use digest::Digest;
+use hmac::Hmac;
+use mac::Mac;
+
+fn calculate_block<M: Mac>(mac: &mut M, info: &[u8], n: u8, scratch: &mut [u8], output: &mut [u8]) {
+    if n != 1 {
+        mac.input(scratch);
+    }
+    let nbuf = [n];
+    mac.input(info);
+    mac.input(&nbuf);
+    mac.raw_result(scratch);
+    mac.reset();
+    copy_memory(output, scratch); // Sigh. :(
+}
+
+/**
```

Comment:
doc comments in rust use ///'s not /**  see [this](http://doc.rust-lang.org/book/documentation.html)